src/utils/cache.h

```diff
@@ -115,6 +117,8 @@ class LruCache {
   // Unpins the element given key and value. Use of LruCacheLock is recommended
   // to automate this pin management.
   void Unpin(K key, V* value) {
+    if (capacity_ == 0) return;
```

Comment:
I think it's better not to do that for unpin.
If cache has capacity of 0 during entire lifetime, then Unpin will never be called, and there's no harm in not having the shortcut.

However, in this (allowed but unlikely) situation we'll have a leak:
1. capacity is non-zero
2. Some entry is pinned
3. Capacity is reset to 0
4. Entry is unpinned.

We used to allow cache size change during the search, that made this situation real. Currently I believe we only do that between moves, but still if it's easy to support, we should.

Reply:
Makes sense. I'll remove the short circuit here.